File: Tarea13_HashSort/HashSort.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class HashSort {
    
    public static void displayArr(List<Integer> inputArr) {
        for (int k : inputArr) {
            System.out.print(k + " ");
        }
        System.out.println();
    }

    public void sort(List<Integer> inputArr) {
        int size = inputArr.size();
        
        for (int gapsize = size / 2; gapsize > 0; gapsize /= 2) {
            for (int j = gapsize; j < size; j++) {
                int val = inputArr.get(j);
                int k = j;
                while (k >= gapsize && inputArr.get(k - gapsize) > val) {
                    inputArr.set(k, inputArr.get(k - gapsize));
                    k = k - gapsize;
                }
                inputArr.set(k, val);
            }
        }
    }

    public static void main(String[] args) {
        List<Integer> inputArr = new ArrayList<>(Arrays.asList(36, 34, 43, 11, 15, 20, 28, 45));
        System.out.println("Arreglo antes de HashSort(Shell) ");
        HashSort.displayArr(inputArr);

        HashSort obj = new HashSort();
        obj.sort(inputArr);

        System.out.println("Arreglo después HashSort(Shell) ");
        HashSort.displayArr(inputArr);
    }
}

File: Tarea13_HashSort/HashSort.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class ShellSort
{
    public static void DisplayArr(List<int> inputArr)
    {
        foreach (int k in inputArr)
        {
            Console.Write(k + " ");
        }
        Console.WriteLine();
    }

    public void Sort(List<int> inputArr)
    {
        int size = inputArr.Count;
        
        for (int gapsize = size / 2; gapsize > 0; gapsize /= 2)
        {
            for (int j = gapsize; j < size; j++)
            {
                int val = inputArr[j];
                int k = j;
                while (k >= gapsize && inputArr[k - gapsize] > val)
                {
                    inputArr[k] = inputArr[k - gapsize];
                    k = k - gapsize;
                }
                inputArr[k] = val;
            }
        }
    }

    public static void Main(string[] args)
    {
        List<int> inputArr = new List<int> { 36, 34, 43, 11, 15, 20, 28, 45 };
        Console.WriteLine("Arreglo antes de HashSort(Shell) ");
        ShellSort.DisplayArr(inputArr);

        ShellSort obj = new ShellSort();
        obj.Sort(inputArr);

        Console.WriteLine("Arreglo después HashSort(Shell) ");
        ShellSort.DisplayArr(inputArr);
    }
}